index.js

```diff
@@ -1,99 +1,50 @@
 
+const R = require('ramda')
 
-const { DepGraph }       = require('dependency-graph');
-const Type               = require('./lib/type');
-const ExpressConfig      = require('./lib/express-config');
-const { OverwriteError } = require('./lib/error');
 
+const throwErrorIfNoSpec = R.when(
+  R.either(R.isEmpty, R.isNil)
+, () => { throw new Error('Spec object is required') }
+)
 
-function validateNode(node) {
 
-  if (typeof node === 'string')
-    throw new TypeError('Dependency node cannot be a string');
+// const throwNotInSpec = (key) => {
+//   throw new Error(`Dependency does not exist: ${key}`)
+// }
 
-  return node;
 
-}
-
-function Yaldic({ allow_overwrite = false } = {}) {
-
-  const graph = new DepGraph();
-
-
-  return {
+const throwErrorIfNotInSpec = (spec) => R.when(
```

Comment:
You should use the [props-check](https://www.npmjs.com/package/props-check) module to output an error when the user asks for something not in the spec.